tests/unittests/tests-gnrc_ipv6_nib/tests-gnrc_ipv6_nib-internal.c

```diff
@@ -964,6 +965,735 @@ static void test_nib_drl_get_dr__success4(void)
+{
+    ipv6_addr_t next_hop = { .u64 = { { .u8 = LINK_LOCAL_PREFIX },
+                                    { .u64 = TEST_UINT64 } } };
+    ipv6_addr_t pfx = { .u64 = { { .u8 = GLOBAL_PREFIX } } };
+    unsigned pfx_len = GLOBAL_PREFIX_LEN;
+
+    for (int i = 0; i < MAX_NUMOF; i++) {
+        _nib_offl_entry_t *dst;
+
+        TEST_ASSERT_NOT_NULL((dst = _nib_offl_alloc(&next_hop, IFACE, &pfx,
+                                                    pfx_len)));
+        TEST_ASSERT_NOT_NULL(dst->next_hop);
+        dst->mode |= _FT;
+        next_hop.u64[1].u64++;
+        pfx.u16[0].u16++;
+        pfx_len--;
+    }
+    TEST_ASSERT_NULL(_nib_offl_alloc(&next_hop, IFACE, &pfx, pfx_len));
+}
+
+/*
+ * Creates MAX_NUMOF off-link entries with different prefixes and different
+ * interfaces and then tries to add another.
+ * Expected result: should return NULL
+ */
+static void test_nib_offl_alloc__no_space_left_diff_iface_pfx_pfx_len(void)
+{
+    static const ipv6_addr_t next_hop = { .u64 = { { .u8 = LINK_LOCAL_PREFIX },
+                                                 { .u64 = TEST_UINT64 } } };
+    ipv6_addr_t pfx = { .u64 = { { .u8 = GLOBAL_PREFIX } } };
+    unsigned pfx_len = GLOBAL_PREFIX_LEN, iface = IFACE;
+
+    for (int i = 0; i < MAX_NUMOF; i++) {
+        _nib_offl_entry_t *dst;
+
+        TEST_ASSERT_NOT_NULL((dst = _nib_offl_alloc(&next_hop, iface, &pfx,
+                                                    pfx_len)));
+        TEST_ASSERT_NOT_NULL(dst->next_hop);
+        dst->mode |= _FT;
+        iface++;
+        pfx.u16[0].u16++;
+        pfx_len--;
+    }
+    TEST_ASSERT_NULL(_nib_offl_alloc(&next_hop, iface, &pfx, pfx_len));
+}
+
+/*
+ * Creates MAX_NUMOF off-link entries with different prefixes, different
+ * interfaces, and different next hop addresses and then tries to add another.
+ * Expected result: should return NULL
+ */
+static void test_nib_offl_alloc__no_space_left_diff_next_hop_iface_pfx_pfx_len(void)
+{
+    ipv6_addr_t next_hop = { .u64 = { { .u8 = LINK_LOCAL_PREFIX },
+                                      { .u64 = TEST_UINT64 } } };
+    ipv6_addr_t pfx = { .u64 = { { .u8 = GLOBAL_PREFIX } } };
+    unsigned pfx_len = GLOBAL_PREFIX_LEN, iface = IFACE;
+
+    for (int i = 0; i < MAX_NUMOF; i++) {
+        _nib_offl_entry_t *dst;
+
+        TEST_ASSERT_NOT_NULL((dst = _nib_offl_alloc(&next_hop, iface, &pfx,
+                                                    pfx_len)));
+        TEST_ASSERT_NOT_NULL(dst->next_hop);
+        dst->mode |= _FT;
+        next_hop.u64[1].u64++;
+        iface++;
+        pfx.u16[0].u16++;
+        pfx_len--;
+    }
+    TEST_ASSERT_NULL(_nib_offl_alloc(&next_hop, iface, &pfx, pfx_len));
+}
+
+/*
+ * Creates MAX_NUMOF off-link entries with different prefixes, different
+ * interfaces, and different next hop addresses and then tries to add another
+ * equal to the last.
+ * Expected result: should return not NULL (the last)
+ */
+static void test_nib_offl_alloc__success_duplicate(void)
```

Comment:
There are subtle differences that are not easily parameterized out. Ideally, this API shouldn't change much.